### DISCUSSION TITLE
RDS migration: proxy mode?

### DISCUSSION THREAD

**Discussion by georgeboot** [planetscale/discussion#328] [Q&A] [answered]
We've started a migration from RDS.

Initial import went smooth and we are now in replica mode (2).

How do we now progress from here? Our workload still speaks to RDS. We basically need to change them to PlanetScale and at the same moment, switch Planetscale to Primary mode (3).

In a podcast I heard something about a feature where PlanetScale proxies the RDS connection so they can basically flip both switches at one.

How do I activate that?

**Comment by orware** [comment, marked as answer]
@georgeboot,

It sounds like you're in the correct spot to be making the transition!

Since you're in Replica Mode already, this is already the state that you described and is what allows for you to start transitioning over to using the PlanetScale hostname and username/password for your database.

It's described in more detail within our [Database Imports](https://planetscale.com/docs/concepts/database-imports#step-2-replica-mode) documentation, but in short, while you are in Replica Mode, our Vitess-powered Schema Routing Rules will allow any writes sent to your PlanetScale database over to your RDS database, and then have the changes synced over to the PlanetScale database automatically.

This allows you to slowly move over all of your existing connections utilized by your workload to the new PlanetScale credentials.

Once you know you have fully transitioned your workload to using the PlanetScale credentials, you can then safely switch to using your PlanetScale database in Primary Mode.

The big caveat I want to share though is that you need to be extra sure that your workload is fully transitioned prior to switching over to Primary Mode.

If you don't it can lead to a "split-brain" situation since those Schema Routing Rules aren't able to be utilized anymore. If you were to accidentally send writes into your RDS database at this point, those aren't able to be synced back into your PlanetScale database since it thinks it is now the primary and doesn't need to, and RDS doesn't have the built-in Vitess functionality for those Schema Routing Rules to work in reverse once the PlanetScale database is the primary. 

I hope that explanation wasn't too confusing, but basically the current Replica Mode you are in is the right one for you to start slowly "flipping the switch" over in each of the pieces of your workload that connects to the database to use the PlanetScale credentials, and once that has been fully completed, switching over to Primary Mode should be seamless since your workload is already utilizing the new credentials.
